htdocs/contact/list.php

```diff
@@ -337,10 +352,13 @@
 if (strlen($search_phone_pro))      $sql.= natural_search('p.phone', $search_phone_pro);
 if (strlen($search_phone_mobile))   $sql.= natural_search('p.phone_mobile', $search_phone_mobile);
 if (strlen($search_fax))            $sql.= natural_search('p.fax', $search_fax);
-if (strlen($search_skype))          $sql.= natural_search('p.skype', $search_skype);
-if (strlen($search_twitter))        $sql.= natural_search('p.twitter', $search_twitter);
-if (strlen($search_facebook))       $sql.= natural_search('p.facebook', $search_facebook);
-if (strlen($search_linkedin))       $sql.= natural_search('p.linkedin', $search_linkedin);
+if (! empty($conf->socialnetworks->enabled)) {
+	foreach ($socialnetworks as $key => $value) {
+		if ($value['active'] && strlen($search_{$key})) {
+			$sql.= natural_search("p.socialnetworks->'$.".$key."'", $search_{$key});
```

Comment:
According to this page http://www.postgresqltutorial.com/postgresql-json/, we should be able to make a 

where p.socialnetworks->facebook LIKE '%criteria%'

So a natural_search("p.socialnetworks->".$key."''",  ...
Why do you have here "p.socialnetworks->'$.".$key."'", ...   (what is goal of the $. ?)

Reply:
I found some info here https://www.sitepoint.com/use-json-data-fields-mysql-databases/
![image](https://user-images.githubusercontent.com/3624836/68092040-544ec800-fe87-11e9-9f5a-eb95254b1296.png)

Reply:
![image](https://user-images.githubusercontent.com/3624836/68092081-c7583e80-fe87-11e9-921b-71b4662b1cb4.png)

Reply:
Tested with mariadb ubuntu 18.04, I got sql syntax errors.